app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/SimilarHandler.kt

```diff
@@ -1,51 +1,61 @@
 import eu.kanade.tachiyomi.data.database.models.Manga
-import eu.kanade.tachiyomi.data.database.models.MangaSimilarImpl
-import eu.kanade.tachiyomi.data.preference.PreferencesHelper
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.NetworkHelper
+import eu.kanade.tachiyomi.network.asObservableSuccess
 import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.handlers.serializers.SimilarMangaResponse
 import eu.kanade.tachiyomi.source.online.utils.MdUtil
 import eu.kanade.tachiyomi.v5.db.V5DbHelper
 import eu.kanade.tachiyomi.v5.db.V5DbQueries
+import okhttp3.CacheControl
+import okhttp3.Request
+import okhttp3.Response
 import rx.Observable
-import uy.kohesive.injekt.Injekt
-import uy.kohesive.injekt.api.get
 import uy.kohesive.injekt.injectLazy
 
 class SimilarHandler {
 
-    val preferences: PreferencesHelper by injectLazy()
-    val v5DbHelper: V5DbHelper by injectLazy()
+    private val network: NetworkHelper by injectLazy()
+    private val v5DbHelper: V5DbHelper by injectLazy()
 
     /**
      * fetch our similar mangas
      */
     fun fetchSimilar(manga: Manga): Observable<MangasPage> {
+        return network.client.newCall(similarMangaRequest(manga))
+                .asObservableSuccess()
+                .map { response ->
+                    similarMangaParse(manga, response)
+                }
+    }
 
-        // Parse the Mangadex id from the URL
-        val mangaid = MdUtil.getMangaId(manga.url).toLong()
-        val lowQualityCovers = preferences.lowQualityCovers()
-        val cacheResponse = preferences.useCacheSource()
-
-        // Get our current database
-        val db = Injekt.get<DatabaseHelper>()
-        val similarMangaDb = db.getSimilar(mangaid).executeAsBlocking() ?: return Observable.just(MangasPage(mutableListOf(), false))
+    private fun similarMangaRequest(manga: Manga): Request {
+        val tempUrl = MdUtil.similarBaseApi+MdUtil.getMangaId(manga.url)+".json"
+        return GET(tempUrl, network.headers, CacheControl.FORCE_NETWORK)
+    }
 
-        // Check if we have a result
-        val similarMangaTitles = similarMangaDb.matched_titles.split(MangaSimilarImpl.DELIMITER)
-        val similarMangaIds = similarMangaDb.matched_ids.split(MangaSimilarImpl.DELIMITER)
-        val similarMangas = similarMangaIds.mapIndexed { index, similarId ->
+    private fun similarMangaParse(manga: Manga, response: Response): MangasPage {
+        if (response.code == 404) {
+            return MangasPage(emptyList(), false)
+        }
+        if (response.isSuccessful.not()) {
+            throw Exception("Error getting search manga http code: ${response.code}")
+        }
+        // TODO: also filter based on the content rating here?
+        // TODO: also append here the related manga?
+        val mlResponse = MdUtil.jsonParser.decodeFromString(SimilarMangaResponse.serializer(), response.body!!.string())
```

Comment:
MdUtil.jsonParser.decodeFromString<SimilarMangaResponse>(response.body!!.string())

I didn't know the above kotlin extension existed before